docs/building-apps/moneygram-access-integration-guide.mdx

```diff
@@ -247,20 +295,84 @@ In each case, the transaction submitted to Stellar must have a memo attached to
 
 ### Poll Until MoneyGram is Ready
 
-Before the application can send funds or instruct the user to provide cash to a MoneyGram agent, the application should confirm with MoneyGram’s server that the transaction is ready to proceed.
+Before the application can send funds or instruct the user to provide cash to a MoneyGram agent, the application should confirm with MoneyGram’s server that the transaction is ready to proceed which is signaled by the `pending_user_transfer_start` status.
```

Comment:
Q: can/should we recommend the use of MoneyGram's webhooks now? When this tutorial was originally created they didn't have them. Now I would imagine this the preferred approach.

Reply:
Nice catch.

Do we have any existing documentation explaining how to setup/integrate MGI's webhook?

Reply:
I don't believe so, @Ifropc @nirojapanda are you familiar with any docs?

Reply:
No, sorry, I don't think there are any docs for that. Their implementation also doesn't match the standard precisely BTW

Reply:
Ok, so do we still want to recommend this approach? If so, can you help create that documentation if you're familiar with the differences?

Reply:
I think it's what Vibrant is doing: only listen to postMessage for closing button, and do NOT listen to the message returned by it, @CassioMG am I correct?

Reply:
Correct, Vibrant only listen to postMessage for closing the button. But it only does that when the `transaction` that comes inside the event message is in a `pending` status to avoid closing the webview for ANY other random postMessage event that may happen during the interactive flow for whatever reason.
Like this:
```
    // If we get a postMessage and the status is any pending one,
    // let's interpret that as a signal to close the window and
    // take user back to the wallet experience
    if (eventMessage.transaction?.status?.includes("pending")) {
      handleClose();
    }
```

Does that make sense?

Sorry about the late response I was OOO these past days

Reply:
Ok, it sounds like using MGI's webhooks instead of postmessages may not be an option yet so we need to continue documenting the postmessage approach. @CassioMG if you think we need to instruct wallets to only close when the status is `pending_user_transfer_start` that is fine by me

Reply:
https://github.com/stellar/stellar-docs/pull/658/commits/960e7cc8617115e256fe08e280c5c72d3592a9de